src/extension-runners/chromium.js

```diff
@@ -0,0 +1,286 @@
+      host: 'localhost',
+    });
+
+    // Prevent unhandled socket error (e.g. when chrome
+    // is exiting, See https://github.com/websockets/ws/issues/1256).
+    this.wss.on('connection', function(socket) {
+      socket.on('error', (err) => {
+        log.debug(`websocket connection error: ${err}`);
+      });
+    });
+
+    // Create the extension that will manage the addon reloads
+    this.reloadManagerExtension = await this.createReloadManagerExtension();
+
+    // Start chrome pointing it to a given profile dir
+    const extensions = [this.reloadManagerExtension].concat(
+      this.params.extensions.map(({sourceDir}) => sourceDir)
+    ).join(',');
+
+    const {chromiumBinary} = this.params;
+
+    log.debug('Starting Chromium instance...');
+
+    if (chromiumBinary) {
+      log.debug(`(chromiumBinary: ${chromiumBinary})`);
+    }
+
+    const chromeFlags = [`--load-extension=${extensions}`];
+
+    if (this.params.args) {
+      chromeFlags.push(...this.params.args);
+    }
+
+    if (this.params.chromiumProfile) {
+      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    }
+
+    let startingUrl;
+    if (this.params.startUrl) {
+      const startingUrls = Array.isArray(this.params.startUrl) ?
+        this.params.startUrl : [this.params.startUrl];
+      startingUrl = startingUrls.shift();
+      chromeFlags.push(...startingUrls);
+    }
+
+    this.chromiumInstance = await this.chromiumLaunch({
+      enableExtensions: true,
+      chromePath: chromiumBinary,
+      chromeFlags,
+      startingUrl,
+    });
+
+    this.chromiumInstance.process.once('close', () => {
+      this.chromiumInstance = null;
+
+      if (!this.exiting) {
+        log.info('Exiting on Chromium instance disconnected.');
+        this.exit();
+      }
+    });
+  }
+
+  wssBroadcast(data: Object) {
+    for (const client of this.wss.clients) {
+      if (client.readyState === WebSocket.OPEN) {
+        client.send(JSON.stringify(data));
+      }
+    }
+  }
+
+  async createReloadManagerExtension() {
+    const tmpDir = new TempDir();
+    await tmpDir.create();
+    this.registerCleanup(() => tmpDir.remove());
+
+    const extPath = path.join(
+      tmpDir.path(),
+      `reload-manager-extension-${Date.now()}`
+    );
+
+    log.debug(`Creating reload-manager-extension in ${extPath}`);
+
+    await asyncMkdirp(extPath);
+
+    await fs.writeFile(
+      path.join(extPath, 'manifest.json'),
+      JSON.stringify({
+        manifest_version: 2,
+        name: 'web-ext Reload Manager Extension',
+        version: '1.0',
+        permissions: ['management', 'tabs'],
+        background: {
+          scripts: ['bg.js'],
+        },
+      })
+    );
+
+    const wssInfo = this.wss.address();
+
+    const bgPage = `(function bgPage(webExtParams) {
+      async function getAllDevExtensions() {
+        const [managerExtension, allExtensions] = await Promise.all([
+          new Promise(r => chrome.management.getSelf(r)),
```

Comment:
There is no need to use `chrome.management.getSelf()` if all you're interested in is the extension ID - use `chrome.runtime.id` instead.

Reply:
:+1: